service-clients-gen/src/main/java/com/flipkart/poseidon/serviceclients/generator/Generator.java

```diff
@@ -88,14 +88,24 @@ private static boolean validateInput(String[] args) {
 
         determineVersion(args[1]);
         String majorVersion = "v" + version.getMajor();
-        packageName = PACKAGE_NAME + moduleName + "." + majorVersion;
+
+        String serviceClientGroupId = System.getProperty("service.clients.groupId");
+        if (serviceClientGroupId == null || serviceClientGroupId.isEmpty()) {
+            serviceClientGroupId = DEFAULT_SERVICE_CLIENT_GROUP_ID;
+        }
+
+        packageName = trimmedPackageName(serviceClientGroupId) + "." + moduleName + "." + majorVersion;
 
         if (args.length > 2) {
             pojoOrdering = args[2].split(",");
         }
         return true;
     }
 
+    private static String trimmedPackageName(String pkgName){
+        return pkgName.endsWith(".") ? pkgName.substring(0, pkgName.length() - 2) : pkgName;
```

Comment:
We should have some validation here. This might throw IndexOutOfBoundsException if user passes just dot (".")

Reply:
added validation on groupId before calling this function.